generator-jhipster-gomicro/generators/server/templates/go/go/handler/db.go

```diff
@@ -0,0 +1,314 @@
+		return err
+	}
+	db.Exec(fmt.Sprintf(stmt, tableName, tableName, tableName))
+	m := req.Record.AsMap()
+	id := req.Id
+	if len(id) == 0 {
+		if mid, ok := m[idKey].(string); ok {
+			id = mid
+		} else {
+			id = uuid.New().String()
+			m[idKey] = id
+		}
+	}
+	bs, _ := json.Marshal(m)
+	err = db.Table(tableName).Create(&Record{
+		ID:   id,
+		Data: bs,
+	}).Error
+	if err != nil {
+		return err
+	}
+
+	// set the response id
+	rsp.Id = id
+	logger.Infof("hii error5")
+
+	return nil
+}
+
+func (e *Db) Update(ctx context.Context, req *db.UpdateRequest, rsp *db.UpdateResponse) error {
+	if len(req.Record.AsMap()) == 0 {
+		return errors.BadRequest("db.update", "missing record")
+	}
+	tableName :="temp"
+	logger.Infof("Updating table '%v'", tableName)
+	db, err := gorm.Open(postgres.Open("postgresql://go@localhost:5433/postgres"), &gorm.Config{})   
+	if err != nil {
+		return err
+	}
+	m := req.Record.AsMap()
+
+	id := req.Id
+	if len(id) == 0 {
+		var ok bool
+		id, ok = m[idKey].(string)
+		if !ok {
+			return fmt.Errorf("update failed: missing id")
+		}
+	}
+
+	return db.Transaction(func(tx *gorm.DB) error {
+		rec := []Record{}
+		err = tx.Table(tableName).Where("id = ?", id).Find(&rec).Error
+		if err != nil {
+			return err
+		}
+		if len(rec) == 0 {
+			return fmt.Errorf("update failed: not found")
+		}
+		old := map[string]interface{}{}
+		err = json.Unmarshal(rec[0].Data, &old)
+		if err != nil {
+			return err
+		}
+		for k, v := range m {
+			old[k] = v
+		}
+		bs, _ := json.Marshal(old)
+
+		return tx.Table(tableName).Save(&Record{
+			ID:   id,
+			Data: bs,
+		}).Error
+	})
+}
+
+func (e *Db) Read(ctx context.Context, req *db.ReadRequest, rsp *db.ReadResponse) error {
+	recs := []Record{}
+    tableName :="temp"
+	db, err := gorm.Open(postgres.Open("postgresql://go@localhost:5433/postgres"), &gorm.Config{})   
+	if err != nil {
+		return err
+	}
+	db = db.Table(tableName)
+	if req.Id != "" {
+		logger.Infof("Query by id: %v", req.Id)
+		db = db.Where("id = ?", req.Id)
+	} 
+	err = db.Debug().Find(&recs).Error
+	if err != nil {
+		return err
+	}
+
+	rsp.Records = []*structpb.Struct{}
+	for _, rec := range recs {
+		m, err := rec.Data.MarshalJSON()
+		if err != nil {
+			return err
+		}
+
+		ma := map[string]interface{}{}
+		json.Unmarshal(m, &ma)
+
+		m, _ = json.Marshal(ma)
+		s := &structpb.Struct{}
+
+		if err = s.UnmarshalJSON(m); err != nil {
+			return err
+		}
+		rsp.Records = append(rsp.Records, s)
+	}
+
+	return nil
+}
+
+func (e *Db) Delete(ctx context.Context, req *db.DeleteRequest, rsp *db.DeleteResponse) error {
+	if len(req.Id) == 0 {
+		return errors.BadRequest("db.delete", "missing id")
+	}
+	tableName :="temp"
+	logger.Infof("Deleting from table '%v'", tableName)
+	db, err := gorm.Open(postgres.Open("postgresql://go@localhost:5433/postgres"), &gorm.Config{})   
+	if err != nil {
+		return err
+	}
+	return db.Table(tableName).Delete(Record{
+		ID: req.Id,
+	}).Error
+}
+<%_ } _%>
+
+<%_ if (mongodb){ _%> 
```

Comment:
Lets stick with PostgreSQL for now, 
keep the mongo code as it is,
first lets only focus on PostgreSQL !